pkg/signup/service/signup_service.go

```diff
@@ -196,20 +197,25 @@ func (s *ServiceImpl) createUserSignup(ctx *gin.Context) (*v1alpha1.UserSignup,
 }
 
 // reactivateUserSignup reactivates the deactivated UserSignup resource with the specified username and userID
-func (s *ServiceImpl) reactivateUserSignup(ctx *gin.Context, existing v1alpha1.UserSignup) (*v1alpha1.UserSignup, error) {
+func (s *ServiceImpl) reactivateUserSignup(ctx *gin.Context, existing *v1alpha1.UserSignup) (*v1alpha1.UserSignup, error) {
 	// Update the existing usersignup's spec and annotations/labels by new values from a freshly generated one.
 	// We don't want to deal with merging/patching the usersignup resource
 	// and just want to reset the spec and annotations/labels so they are the same as in a freshly created usersignup resource.
 	newUserSignup, err := s.newUserSignup(ctx)
 	if err != nil {
 		return nil, err
 	}
+	log.WithValues(map[string]interface{}{"toolchain.dev.openshift.com/activation-counter": existing.Annotations["toolchain.dev.openshift.com/activation-counter"]}).
```

Comment:
it's just a detail, but couldn't we use a constant?:
```suggestion
	log.WithValues(map[string]interface{}{v1alpha1.UserSignupActivationCounterAnnotationKey: existing.Annotations[v1alpha1.UserSignupActivationCounterAnnotationKey]}).
```